index.js

```diff
@@ -426,6 +431,21 @@ class WebTorrent extends EventEmitter {
     args[0] = `[${this._debugId}] ${args[0]}`
     debug(...args)
   }
+
+  _getByHash (infoHashHash) {
+    var i, torrent
+    var len = this.torrents.length
+    var hReq2 = Buffer.from('req2', 'utf8').toString('hex')
+
+    for (i = 0; i < len; i++) {
+      torrent = this.torrents[i]
+      if (infoHashHash === sha1.sync(Buffer.from(hReq2 + torrent.infoHash, 'hex'))) {
```

Comment:
This is a perf killer, iterate over all torrents and sha1 >.<, isn't there a better approach?

Reply:
fixed, but it's not pref killer